src/infra/http/validation/RequiredFieldsValidator.ts

```diff
@@ -0,0 +1,18 @@
+import { Either, left, right } from '@core/logic/Either'
+import { MissingParamError } from './errors/MissingParamError'
+import { Validator } from './Validator'
+
+export class RequiredFieldsValidator<T = any> implements Validator<T> {
+  public validate(data: T): Either<MissingParamError, null> {
+    const fields = Object.getOwnPropertyNames(data)
+    for (const field of fields) {
+      if (data[field] !== false && !data[field]) {
+        return left(new MissingParamError(field))
+      }
+    }
+
+    return right(null)
+
+    // TODO: validate if a field is an object with "sub-properties", and if this object has required properties
```

Comment:
Can you create an issue for this one and remove the comment?

Reply:
Sure. I was reviewing the controllers, and I'm not sure if we'll actually need this feature, but it's good to have this "problem" already solved, this way, if in the future a controller needs to receive an object with required properties inside the body, the validator well be able to validate it.

Reply:
I think we can create the feature as soon as it's needed. We don't need to care about it now.

Reply:
Perfect.